exchange/bilaxy/api.go

```diff
@@ -0,0 +1,514 @@
+	coinsData := make(map[string]*CoinsData)
+	strRequestUrl := "/v1/currencies"
+	strUrl := API_URL + strRequestUrl
+
+	jsonCurrencyReturn := exchange.HttpGetRequest(strUrl, nil)
+	if err := json.Unmarshal([]byte(jsonCurrencyReturn), &coinsData); err != nil {
+		return fmt.Errorf("%s Get Coins Json Unmarshal Err: %v %v", e.GetName(), err, jsonCurrencyReturn)
+	}
+
+	for _, data := range coinsData {
+		c := &coin.Coin{}
+		switch e.Source {
+		case exchange.EXCHANGE_API:
+			c = coin.GetCoin(data.Currency)
+			if c == nil {
+				c = &coin.Coin{}
+				c.Code = data.Currency
+				c.Name = data.FullName
+				//c.Website = data.URL
+				//c.Explorer = data.BlockURL
+				coin.AddCoin(c)
+			}
+		case exchange.JSON_FILE:
+			c = e.GetCoinBySymbol(data.Currency)
+		}
+
+		if c != nil {
+			//confirmation, _ := strconv.Atoi(data.ConfirmTimes)
+			coinConstraint := e.GetCoinConstraint(c)
+			txFee, _ := strconv.ParseFloat(data.FixedWithdrawFee, 64)
+			if coinConstraint == nil {
+				coinConstraint = &exchange.CoinConstraint{
+					CoinID:       c.ID,
+					Coin:         c,
+					ExSymbol:     data.Currency,
+					ChainType:    exchange.MAINNET,
+					TxFee:        txFee,
+					Withdraw:     data.WithdrawEnabled,
+					Deposit:      data.DepositEnabled,
+					Confirmation: DEFAULT_CONFIRMATION,
+					Listed:       true,
+				}
+			} else {
+				coinConstraint.ExSymbol = data.Currency
+				coinConstraint.TxFee = txFee
+				coinConstraint.Withdraw = data.WithdrawEnabled
+				coinConstraint.Deposit = data.DepositEnabled
+				coinConstraint.Confirmation = DEFAULT_CONFIRMATION
+			}
+
+			e.SetCoinConstraint(coinConstraint)
+		}
+	}
+	return nil
+}
+
+/* GetPairsData - Get Pairs Information (If API provide)
+Step 1: Change Instance Name    (e *<exchange Instance Name>)
+Step 2: Add Model of API Response
+Step 3: Modify API Path(strRequestUrl)*/
+func (e *Bilaxy) GetPairsData() error {
+	//pairsData := PairsData{}
+	pairsData := make(map[string]*PairsData)
+	strRequestUrl := "/v1/pairs"
+	strUrl := API_URL + strRequestUrl
+
+	jsonSymbolsReturn := exchange.HttpGetRequest(strUrl, nil)
+	if err := json.Unmarshal([]byte(jsonSymbolsReturn), &pairsData); err != nil {
+		return fmt.Errorf("%s Get Pairs Json Unmarshal Err: %v %v", e.GetName(), err, jsonSymbolsReturn)
+	}
+
+	for key, data := range pairsData {
+		p := &pair.Pair{}
+		switch e.Source {
+		case exchange.EXCHANGE_API:
+			base := coin.GetCoin(data.Quote)
+			target := coin.GetCoin(data.Base)
+			if base != nil && target != nil {
+				p = pair.GetPair(base, target)
+			}
+		case exchange.JSON_FILE:
+			p = e.GetPairBySymbol(key)
+		}
+
+		if p != nil {
+			pairConstraint := e.GetPairConstraint(p)
+			if pairConstraint == nil {
+				pairConstraint = &exchange.PairConstraint{
+					PairID:      p.ID,
+					Pair:        p,
+					ExSymbol:    key,
+					MakerFee:    DEFAULT_MAKER_FEE,
+					TakerFee:    DEFAULT_TAKER_FEE,
+					LotSize:     DEFAULT_LOT_SIZE,
+					PriceFilter: DEFAULT_PRICE_FILTER,
```

Comment:
these two constraints are important, from api fields 'amount_precision' and 'price_precision'
Transform the value into form like '0.01'